### ISSUE TITLE
pywb doesn't work well with WARCs created using wpull

### ISSUE THREAD

**Issue by Svekla** [ikreymer/pywb#175]
When I try to add WARC from wpull some files are missing. For example:

Dump:
[forumpostalsiteinfo-30032016 (1).zip](https://github.com/ikreymer/pywb/files/198276/forumpostalsiteinfo-30032016.1.zip)

File:
http://forum.postalsite.info/templates/PostalSite/images/folder_locked_big.gif

CDX from wpull:
```
http://forum.postalsite.info/templates/PostalSite/images/folder_locked_big.gif 1459376733 - - WQ26USAMZQQ4PC64GS7D2R7577WWSFC7 1092 219055 forumpostalsiteinfo-30032016-00000.warc.gz <urn:uuid:f66de344-8f69-4174-af54-89b603c4f5ac>
```

Record from file:
```
# python -c 'import sys; f = open("forumpostalsiteinfo-30032016-00000.warc.gz"); f.seek(219055); sys.stdout.write(f.read(1092))' | gunzip
WARC/1.0
WARC-Type: response
Content-Type: application/http;msgtype=response
WARC-Date: 2016-03-30T22:25:33Z
WARC-Record-ID: <urn:uuid:f66de344-8f69-4174-af54-89b603c4f5ac>
WARC-Target-URI: http://forum.postalsite.info/templates/PostalSite/images/folder_locked_big.gif
WARC-IP-Address: 85.128.226.175
WARC-Concurrent-To: <urn:uuid:81c72d5e-27ef-43cf-be61-674afc7c66c7>
WARC-Block-Digest: sha1:N4OMUJLKPU7UHZTY2DBE6O7LRU5OFHJ7
WARC-Payload-Digest: sha1:WQ26USAMZQQ4PC64GS7D2R7577WWSFC7
Content-Length: 694
WARC-Warcinfo-ID: <urn:uuid:a01265ae-f1f6-49bf-b498-092f6079b56c>

HTTP/1.1 200 OK
Date: Wed, 30 Mar 2016 22:25:33 GMT
Content-Type: image/gif
Content-Length: 455
Connection: keep-alive
Last-Modified: Tue, 07 Oct 2014 09:43:01 GMT
ETag: "1c7-504d2066d04cc"
Accept-Ranges: bytes
Server: Apache/2

GIF89a((�������|rh]SI?4* �
�O2$�p��`:%,v9�6a ���gDA- ����!����,((��$�di�h��l��p,�t
��                       �?�Y0����)ZG��ܩ�b���ab8��f~8f�xoe�eJ�E�T�Y�^��3:T�
�29�uqZS��>8
}в8�����    q���69�e���78    k��1��K6L8
        qVߙ8|����9Rl
                    ���Ȣ�Y�R`��T~�{�@�&��t���+�'!��@԰i�.�,X
�2u\    �'                                                 �������aA,F`�堂R��DI���
�J*��Y�勦J_s�h4F%��~�
��y���Iz�p�E!5�M�Y�Z�E��_����"�;
```
And in pywb 0.11.4 (importing without cdx):
```
127.0.0.1 - - [31/Mar/2016 19:51:28] "GET /test/20160330222541im_/http://forum.postalsite.info/templates/PostalSite/images/folder_locked_big.gif HTTP/1.1" 404 158
```

Thanks for help.

**Comment by ikreymer** [comment]
I was able to repro this and was rather confused. Turns out this identified a rather obscure bug, which occurs when the 'check for POST requests and merge them into response' flag is enabled. Due to an optimization, it turns out that this doesn't work right when there are multiple contiguous `request` records, which wpull produces. This doesn't happen if request and response records alternate, as is the output of many other tools, but of course not required.

I'll add a fix to an update release.

In the meantime, a quick fix is to regenerate the cdxj by running something like:

`cdx-indexer -j -s WARC.warc.gz > ./collections/test/indexes/index.cdxj`

This should give you a working index file.

(The default `wb-manager` command adds the `-p` flag which is what's causing the issue).

Thanks for submitting this issue!
